libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2533,8 +2533,15 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	// TODO some members might be pure, but for example `address(0x123).balance` is not pure
 	// although every subexpression is, so leaving this limited for now.
 	if (auto tt = dynamic_cast<TypeType const*>(exprType))
+	{
 		if (tt->actualType()->category() == Type::Category::Enum)
 			annotation.isPure = true;
+		if (tt->actualType()->category() == Type::Category::Contract)
+			if (annotation.type->category() == Type::Category::Function)
+				annotation.isPure = true;
```

Comment:
I think I could sleep better with
```suggestion
				annotation.isPure = _memberAccess.expression().annotation().isPure;
```

Reply:
Changed it.